### PR TITLE
revise anariMapFrame() to include validation output values for applications

### LINKED ISSUE #29: Return image specification from `anariMapFrame`
I propose we change the signature of `anariMapFrame` from
```
const void* anariMapFrame(ANARIDevice device, ANARIFrame frame, const char* channel);
```
to
```
const void* anariMapFrame(ANARIDevice device, ANARIFrame frame, const char* channel,
   uint32_t *size, ANARIDataType *type);
```
Which would return the channel dimensions (`uint32_t[2]`) and data type of the returned pointer by writing to the `size` and `type` pointers if they are not `NULL`.

This way the returned data can be correctly used and its layout verified without external knowledge of what parameters were set on the frame object. This particularly comes up in tooling like tracing instrumentation which would otherwise have to externally track the parameters. This gets additionally complicated when extensions add more channels whose associated enabling parameters need to be intrinsically known to the tooling.

This is to my knowledge the only part of the "core" API requiring knowledge of object parameters to write defined C(++) code. By comparison the array API doesn't require inspection of object parameters since the dimensions and types are passed as function arguments during array creation.

### PR DESCRIPTION
**Pull request by jeffamstutz**
Closes #29.

### CODE REVIEW

File: chapters/object_types/frame.txt [hunk 1]
```diff
@@ -75,13 +75,21 @@ access the stored pixel information – via
 
 [source,cpp]
 ....
-const void *anariMapFrame(ANARIDevice, ANARIFrame, const char *channel);
+const void* anariMapFrame(ANARIDevice,
+                          ANARIFrame,
+                          const char* channel,
+                          uint32_t *width,
+                          uint32_t *height,
+                          ANARIDataType *type);
```
Comment (johguenther) [inline chapters/object_types/frame.txt:83]: Better/needed to name it `pixelType`? We have `elementType` for arrays.

File: chapters/object_types/frame.txt [hunk 2]
```diff
@@ -75,13 +75,21 @@ access the stored pixel information – via
...
 ....
 
 Only channels that have been set as parameters to the frame can be
 mapped, the type of the pixels matches the corresponding parameter
 value.
 
+`width`, `height`, and `type` are output parameters for the application to
```
Comment (johguenther) [inline chapters/object_types/frame.txt:90]: It looks odd to start the sentence lower-case, how about prepending "The arguments `width`..."? Could even merge this into the prev. paragraph line 88.

Comment (jeffamstutz) [reply chapters/object_types/frame.txt]: Addressed in latest push, sorry I didn't see this comment!

File: chapters/object_types/frame.txt (hunk 1 above)
Comment (jeffamstutz) [reply chapters/object_types/frame.txt]: Addressed in latest push.

**Pull request merged**